aptos-move/aptos-aggregator/src/aggregator_extension.rs

```diff
@@ -13,22 +16,50 @@ use move_binary_format::errors::{PartialVMError, PartialVMResult};
 use move_core_types::account_address::AccountAddress;
 use std::collections::{BTreeMap, BTreeSet};
 
+/// Describes how the `speculative_start_value` in
```

Comment:
nit: these two comment lines can be merged.
also maybe "was obtained" instead of "is obtained" (also in comments below)